storage/src/main/java/tech/pegasys/teku/storage/store/StoreVoteUpdater.java

```diff
@@ -92,7 +92,10 @@ public Bytes32 applyForkChoiceScoreChanges(
   public void commit() {
     // Votes are applied to the store immediately since the changes to the in-memory ProtoArray
     // can't be rolled back.
-    store.votes.putAll(votes);
+    votes.forEach(
+        (key, value) -> {
+          store.votes[key.intValue()] = value;
+        });
     store.highestVotedValidatorIndex =
         store.highestVotedValidatorIndex.max(
             votes.keySet().stream().max(Comparator.naturalOrder()).orElse(UInt64.ZERO));
```

Comment:
Can probably use a plain for loop instead of forEach to find the maximum changes validator index here too.